src/lib/y2partitioner/widgets/btrfs_subvolumes_table.rb

```diff
@@ -0,0 +1,122 @@
+require "yast"
+require "cwm/table"
+require "y2partitioner/widgets/help"
+
+module Y2Partitioner
+  module Widgets
+    # Table widget to represent the btrfs subvolumes of a specific filesystem
+    class BtrfsSubvolumesTable < CWM::Table
+      include Help
+
+      attr_reader :filesystem
+
+      # @param filesystem [Y2Storage::Filesystems::BlkFilesystem] a btrfs filesystem
+      def initialize(filesystem)
+        textdomain "storage"
+
+        @filesystem = filesystem
+      end
+
+      # Table header
+      def header
+        columns.map { |c| send("#{c}_title") }
+      end
+
+      # Items do not include top level and defatul btrfs subvolumes
+      #
+      # @see Y2Storage::BtrfsSubvolume#top_level?
+      # @see Y2Storage::BtrfsSubvolume#default_btrfs_subvolume?
+      def items
+        subvolumes.map { |s| values_for(s) }
+      end
+
+      # Updates table content
+      def refresh
+        change_items(items)
+      end
+
+      # Returns the selected subvolume
+      #
+      # @return [BtrfsSubvolume, nil] nil if anything is selected
+      def selected_subvolume
+        return nil if items.empty? || !value
+
+        path = value[/table:subvolume:(.*)/, 1]
+        filesystem.find_btrfs_subvolume_by_path(path)
+      end
+
+      # Builds the help, including columns help
+      def help
+        text = []
+
+        text << _("<p>The table contains:</p>")
+
+        columns.each do |column|
+          help_method = "#{column}_help"
+          text << "<p>" + send(help_method) + "</p>" if respond_to?(help_method, true)
+        end
+
+        text.join("\n")
+      end
+
+    private
+
+      def columns
+        [:path, :nocow]
+      end
+
+      def values_for(subvolume)
+        [row_id(subvolume)] + columns.map { |c| send("#{c}_value", subvolume) }
+      end
+
+      def row_id(subvolume)
+        "table:subvolume:#{subvolume.path}"
+      end
+
+      # Top level and default subvolumes should not be listed
+      def subvolumes
+        filesystem.btrfs_subvolumes.select do |subvolume|
+          !subvolume.top_level? && !subvolume.default_btrfs_subvolume?
+        end
+      end
+
+      # FIXME: BtrFS could belong to several devices
+      def device(filesystem)
+        filesystem.plain_blk_devices.first
+      end
+
+      # Column titles
+
+      def path_title
+        # TRANSLATORS: table header, subvolume path e.g. "@/home"
+        _("Path")
+      end
+
+      def nocow_title
+        # TRANSLATORS: table header, nocow subvolume attribute
```

Comment:
👍 for translators help. I just suggest to write what nocow accronym strands for ( do not use copy on write feature )